larpandoracontent/LArThreeDReco/LArHitCreation/ThreeDHitCreationAlgorithm.cc

```diff
@@ -286,13 +507,14 @@ void ThreeDHitCreationAlgorithm::RefineHitPositions(const ThreeDSlidingFitResult
         }
         else
         {
-            std::cout << "ThreeDHitCreationAlgorithm::IterativeTreatment - Unexpected number of trajectory samples" << std::endl;
-            throw StatusCodeException(STATUS_CODE_FAILURE);
+            // std::cout << "ThreeDHitCreationAlgorithm::IterativeTreatment - Unexpected number of trajectory samples" << std::endl;
+            // throw StatusCodeException(STATUS_CODE_FAILURE);
+            continue; // TODO: Fix this, check why it happens and if it is specific to the changes I've made.
         }
 
         double bestY(std::numeric_limits<double>::max()), bestZ(std::numeric_limits<double>::max());
         PandoraContentApi::GetPlugins(*this)->GetLArTransformationPlugin()->GetMinChiSquaredYZ(u, v, w, sigmaU, sigmaV, sigmaW, uFit, vFit, wFit, sigma3DFit, bestY, bestZ, chi2);
-        position3D.SetValues(pCaloHit2D->GetPositionVector().GetX(), static_cast<float>(bestY), static_cast<float>(bestZ));
+        position3D.SetValues(protoHit.GetPosition3D().GetX(), static_cast<float>(bestY), static_cast<float>(bestZ));
```

Comment:
Maybe a slightly odd/worrying change, since it changes the underlying 3D reco, rather than just being my own standalone stuff.

However, for all non-interpolated spacepoints, the calo hit and the proto hit have the exact same X coordinate, so this change should not change anything for non-interpolated spacepoints.

For an interpolated spacepoint, its entirely possible for the calo hit and the proto hit to have different X coordinates....the proto hit is interpolated, so it can have a slightly different X. Normally, not an issue, but when running John's iterative treatment on interpolated spacepoints, without this change every interpolated spacepoint would reset its X coord to some random other X value it didn't start with, causing all the interpolated points to end up in odd positions.

Reply:
As a minimum I think this should be an option; if using RANSAC then use your new version, if not using RANSAC retain the existing SetValues call to avoid any possibility these values could differ in the underlying reco.